docs/guides/moodleapp/development/plugins-development-guide.md

```diff
@@ -0,0 +1,2009 @@
+
+This should suffice for developing plugins. However, if you are working on advanced functionality and you need to run the application from the source code, you can find more information in the [[Moodle App Development Guide]].
+
+### Development workflow
+
+Before getting into the specifics of your plugin, we recommend that you start adding a simple "Hello World" button in the app to see that everything works properly.
+
+Let's say your plugin is called `local_hello`, you can start by adding the following files:
+
+```php title="db/mobile.php"
+<?php
+
+$addons = [
+    'local_hello' => [
+        'handlers' => [
+            'hello' => [
+                'delegate' => 'CoreMainMenuDelegate',
+                'method' => 'view_hello',
+                'displaydata' => [
+                    'title' => 'hello',
+                    'icon' => 'earth',
+                ],
+            ],
+        ],
+        'lang' => [
+            ['hello', 'local_hello'],
+        ],
+    ],
+];
+```
+
+```php title="classes/output/mobile.php"
+<?php
+
+namespace local_hello\output;
+
+defined('MOODLE_INTERNAL') || die();
+
+class mobile {
+
+    public static function view_hello() {
+        return [
+            'templates' => [
+                [
+                    'id' => 'main',
+                    'html' => '<h1 class="text-center">{{ "plugin.local_hello.hello" | translate }}</h1>',
+                ],
+            ],
+        ];
+    }
+
+}
+```
+
+```php title="lang/en/local_hello.php"
+<?php
+
+$string['hello'] = 'Hello World';
+```
+
+Once you've done that, try logging into your site in the app and you should see a new button in the main menu or more menu (depending on the device) saying "Hello World". If you press this button, you should see a page saying "Hello World!".
+
+Congratulations, you have written your first Moodle plugin with moodle support!
+
+You can read the rest of this page to learn more about mobile plugins and start working on your plugin. Here's some things to keep in mind:
+
+- If you change the `mobile.php` file, you will have to refresh the browser. And remember to [disable the network cache](https://developer.chrome.com/docs/devtools/network/reference/#disable-cache).
+- If you change an existing template or function, you won’t have to refresh the browser. In most cases, doing a PTR (Pull To Refresh) in the page that displays the template will suffice.
+- If any of these doesn't show your changes, you may need to [purge all caches](https://docs.moodle.org/311/en/Developer_tools#Purge_all_caches) to avoid problems with the auto-loading cache.
```

Comment:
```suggestion
- If any of these doesn't show your changes, you may need to [purge all caches](https://docs.moodle.org/en/Developer_tools#Purge_all_caches) to avoid problems with the auto-loading cache.
```

Reply:
It's better to use the generic URL that will redirect to the latest Moodle version :-)